modules/router/src/services/autoRebalance.ts

```diff
@@ -179,87 +202,188 @@ export const rebalanceIfNeeded = async (
   // on each rebalance interval, check rebalance thresholds for allowedSwaps
   // if out of threshold, check if active rebalance for that swap exists
   // if it does not, create one and go through the approve/execute flow
-  // if it does exist, check status. if status == {completed: true} and if there is a tx in the response (is the case with matic withdraw), send the tx, and mark as completed. if no tx in the response, mark as completed immediately
-  // if it is not completed, wait for next poll
+  // if it does exist, check status. if status == {completed: true} and/or
+  // if there is a tx in the response (is the case with matic withdraw),
+  // send the tx, and mark as completed. if no tx in the response, mark as completed immediately
+  // if it is not finished, wait for next poll
 
   // check if an active rebalance is in progress
-  let latest = await store.getLatestRebalance(swap);
-  console.log("latest: ", latest);
-  if (!latest) {
-    // set dummy value
-    latest = {
-      status: RouterRebalanceStatus.COMPLETE,
+  let latestRebalance = await store.getLatestRebalance(swap);
+
+  if (!latestRebalance || latestRebalance.status === RouterRebalanceStatus.COMPLETE) {
```

Comment:
Actually the way that I have it, we should no longer need the `FINISHED` status.

If rebalance completed, but completion tx needed, and there's no receipt/confirmation yet - it won't transition to COMPLETE status. It will save the completion tx hash, but wait for receipt before going to that last state.

Reply:
Got it, then let's take it out